app/models/miq_queue.rb

```diff
@@ -117,6 +117,9 @@ def self.put(options)
     options[:task_id]      = $_miq_worker_current_msg.try(:task_id) unless options.key?(:task_id)
     options[:role]         = options[:role].to_s unless options[:role].nil?
 
+    # Let's deprecate supplying non-array args soon
+    options[:args] = [options[:args]] if options[:args] && !options[:args].kind_of?(Array)
```

Comment:
It gets be splatted on the send in deliver:  https://github.com/ManageIQ/manageiq/blob/d6c1058e9f8a3ac19ff4452b3df2f446ff5dde84/app/models/miq_queue.rb#L339-L345

But, good point, I'll add or ensure there's a test for that.